File: ppt/src/main/java/com/codecool/ppt/model/Book.java
package com.codecool.ppt.model;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Book {

    @Id
    private String id;
    private String name;
    private Author author;
    private String description;
    private String imageUrl;
    private int numberOfPages;

}

File: frontend/src/types/Book.ts
export type Book ={
    name:string,
    author:string,
    description: string,
    imageUrl:string,
    text:any,
    numberOfPages: number
}

export function getEmptyBook(): Book {
    return {
        name:"",
        author:"",
        description: "",
        imageUrl:"",
        text: null,
        numberOfPages: 0
    }
}
